### PR TITLE
[DOC] Add gsub processor documentation

### PR DESCRIPTION
**Pull request by vagimeli**
### Description
Closes content gap

### Issues Resolved
#4647 

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vagimeli** [comment]
@hdhalter We need a dev to take the first pass at drafting this content. I can support them in refining the content and getting it through the documentation process. The ingest processor template is provided with this PR, so it should help the dev get started. Thanks!

**Comment by vagimeli** [comment]
@gaobinlong This PR is ready for your technical review. Thank you!

**Review by gaobinlong — APPROVED** [review]
LGTM.

File: _ingest-pipelines/processors/gsub.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+|-----------|-----------|-----------|
+`field` | Required | The field to apply the replacement.
+`pattern` | Required | The pattern to be replaced.
+`replacement` | Required | The string to replace the matching patterns with.
+`if` | Optional | A condition for running the processor.
+`ignore_missing` | Optional | If `true` and the specified field does not exist or is null, then the processor will quietly exit without modifying the document.
+`ignore_failure` | Optional | Specifies whether the processor continues execution even if it encounters an error. If set to `true`, then failures are ignored. Default is `false`.
+`on_failure` | Optional | A list of processors to run if the processor fails.
+`tag` | Optional | An identifier tag for the processor. Useful for debugging in order to distinguish between processors of the same type.
+`target_field` | Optional | The name of the field in which to store the parsed data. If not specified, the value is stored in place in the `field` field. Default is `field`.
```
Comment (gaobinlong) [inline _ingest-pipelines/processors/gsub.md:37]: How about moving the `target_field` above the common parameters, I think this parameter is more important than the common parameters.

File: _ingest-pipelines/processors/gsub.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
+---
+layout: default
+title: gsub
+parent: Ingest processors
+nav_order: 130
+---
+
+# Gsub processor
+
+The `gsub` processor performs a regular expression search-and-replace operation on string fields in incoming documents. If the field contains an array of strings, the operation is applied to all elements in the array. However, if the field contains non-string values, the processor throws an exception. Uses cases for the `gsub` processor include removing sensitive information from log messages or user-generated content, normalizing data formats or conventions (for example, converting date formats, removing special characters), and extracting or transforming substrings from field values for further processing or analysis.
```
Comment (natebower) [inline _ingest-pipelines/processors/gsub.md:10]:
```suggestion
The `gsub` processor performs a regular expression search-and-replace operation on string fields in incoming documents. If the field contains an array of strings, the operation is applied to all elements in the array. However, if the field contains non-string values, the processor throws an exception. Use cases for the `gsub` processor include removing sensitive information from log messages or user-generated content, normalizing data formats or conventions (for example, converting date formats, removing special characters), and extracting or transforming substrings from field values for further processing or analysis.
```

File: _ingest-pipelines/processors/gsub.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+```
+{% include copy-curl.html %}
+
+## Configuration parameters
+
+The following table lists the required and optional parameters for the `gsub` processor.
+
+Parameter | Required/Optional | Description |
+|-----------|-----------|-----------|
+`field` | Required | The field to apply the replacement.
```
Comment (natebower) [inline _ingest-pipelines/processors/gsub.md:29]:
```suggestion
`field` | Required | The field to apply the replacement to.
```

File: _ingest-pipelines/processors/gsub.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+`pattern` | Required | The pattern to be replaced.
+`replacement` | Required | The string to replace the matching patterns with.
```
Comment (natebower) [inline _ingest-pipelines/processors/gsub.md:31]:
```suggestion
`replacement` | Required | The string that will replace the matching patterns.
```

File: _ingest-pipelines/processors/gsub.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+`on_failure` | Optional | A list of processors to run if the processor fails.
+`tag` | Optional | An identifier tag for the processor. Useful for debugging in order to distinguish between processors of the same type.
+
+## Using the processor
+
+Follow these steps to use the processor in a pipeline.
+
+### Step 1: Create a pipeline
+
+The following query creates a pipeline named `gsub_pipeline` that uses the `gsub` processor to replace all occurrences of the word `error` with `warning` in the `message` field:
```
Comment (natebower) [inline _ingest-pipelines/processors/gsub.md:45]:
```suggestion
The following query creates a pipeline named `gsub_pipeline` that uses the `gsub` processor to replace all occurrences of the word `error` with the word `warning` in the `message` field:
```

File: _ingest-pipelines/processors/gsub.md [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+PUT logs/_doc/1?pipeline=gsub_pipeline
+{
+  "message": "This is an error message"
+}
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The following response shows that the request indexes the document into the index named `logs` and that the `gsub` processor has replaced all occurrences of the word `error` with `warning` in the `message` field:
```
Comment (natebower) [inline _ingest-pipelines/processors/gsub.md:123]:
```suggestion
The following response shows that the request indexed the document into the index named `logs` and that the `gsub` processor replaced all occurrences of the word `error` with the word `warning` in the `message` field:
```

**Review by natebower — COMMENTED** [review]
@vagimeli Please see my changes and let me know if you have any questions. Thanks!

**Comment by vagimeli** [comment]
@natebower  Thank you for the thorough review. I've accepted your edits. Will you approve for merging? I opened the PR instead of having the dev open it and now can't approve it. Thank you.

**Pull request merged**